elliott/elliottlib/cli/pin_cli.py

```diff
@@ -0,0 +1,324 @@
+
+        # categorize the nvrs
+        art_images_by_comp = {i.get_component_name(): i for i in self.runtime.image_map.values()}
+        art_rpms_by_comp = {r.get_component_name(): r for r in self.runtime.rpm_map.values()}
+
+        images, art_rpms, non_art_rpms, rhcos = [], [], [], []
+        major, minor = self.runtime.get_major_minor()
+        nvrs = set(self.nvrs)
+        for nvr in nvrs:
+            parsed = parse_nvr(nvr)
+            if parsed['name'] in art_images_by_comp:
+                if f"{major}.{minor}" not in parsed['version']:
+                    raise ValueError(f"Does nvr belong to the current group? {nvr}")
+                images.append(nvr)
+            elif parsed['name'].startswith("rhcos"):
+                if f"{major}{minor}" not in parsed['version']:
+                    raise ValueError(f"Does rhcos nvr belong to the current group: {nvr}")
+                rhcos.append(nvr)
+            elif parsed['name'] in art_rpms_by_comp:
+                if f"{major}.{minor}" not in parsed['version']:
+                    raise ValueError(f"Does nvr belong to the current group? {nvr}")
+                art_rpms.append(nvr)
+            else:
+                non_art_rpms.append(nvr)
+
+        if len(rhcos) > 1:
+            raise ValueError(
+                "Did not expect more than one rhcos nvr. Provide only 1 nvr which will be pinned for all arches.")
+
+        nvrs_to_fetch = images + art_rpms + non_art_rpms
+        if nvrs_to_fetch:
+            if self.runtime.build_system == Engine.BREW.value:
+                self.validate_nvrs_in_brew(nvrs_to_fetch)
+            else:
+                # only validate if nvrs were passed in directly
+                if not self.pr:
+                    await self.validate_nvrs_in_konflux_db(nvrs_to_fetch)
+
+        images_changed = self.pin_images(art_images_by_comp, images)
+        art_rpms_changed = self.pin_rpms(art_rpms_by_comp, art_rpms)
+        non_art_rpms_changed = self.pin_non_art_rpms(non_art_rpms)
+        rhcos_changed = self.pin_rhcos(rhcos[0]) if rhcos else False
+
+        assembly_changed = images_changed or art_rpms_changed or non_art_rpms_changed or rhcos_changed
+        if assembly_changed:
+            print(yaml.dump(self.assembly_config.primitive()))
+        else:
+            LOGGER.info("No change in assembly config. Pins already exist.")
+
+    async def get_nvrs_for_pr(self):
+        images, rpms, nvrs = [], [], []
+        repo_url = self.pr.split("/pull")[0]
+        for image in self.runtime.image_map.values():
+            if repo_url == image.config.content.source.git.web:
+                images.append(image)
+        for rpm in self.runtime.rpm_map.values():
+            if repo_url == rpm.config.content.source.git.web:
+                rpms.append(rpm)
+        LOGGER.info(f"ART components building from {repo_url}: {[c.get_component_name() for c in images + rpms]}")
+
+        if not images + rpms:
```

Comment:
Using `+` between 2 lists will actually create a new list, which might be resource-consuming. The following should be better for both performance and readability:
```suggestion
        if not images and not rpms:
```

Reply:
Updated